healthcheck_test.go

```diff
@@ -0,0 +1,99 @@
+package client_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	health "github.com/ONSdigital/dp-healthcheck/healthcheck"
+	nomad "github.com/ONSdigital/dp-nomad"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+type MockedHTTPResponse struct {
+	StatusCode int
+	Body       string
+}
+
+const Name = "nomad"
+
+var ctx = context.Background()
+
+func getMockNomad(expectRequest http.Request, mockedHTTPResponse MockedHTTPResponse) *nomad.Client {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != expectRequest.Method {
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Write([]byte("unexpected HTTP method used"))
+			return
+		}
+		w.WriteHeader(mockedHTTPResponse.StatusCode)
+		fmt.Fprintln(w, mockedHTTPResponse.Body)
+	}))
+
+	client, _ := nomad.NewClient(ts.URL, "testdata/testCertFile", false)
+
+	return client
+}
+
+func TestCheckOk(t *testing.T) {
+	Convey("Given that the vault client is available", t, func() {
+		//c, _ := nomad.NewClient(ts.URL, "", true)
+		mockedNomad := getMockNomad(
+			http.Request{Method: "GET"},
+			MockedHTTPResponse{StatusCode: 200, Body: "{\"status\": \"OK\"}"})
+
+		Convey("When a check is performed", func() {
+			checkState := health.NewCheckState(nomad.ServiceName)
+
+			Convey("Then the Checker returns no error", func() {
+				//err := c.Checker(context.Background(), checkState)
+				err := mockedNomad.Checker(ctx, checkState)
+				So(err, ShouldBeNil)
+				So(checkState.Status(), ShouldEqual, health.StatusOK)
+				So(checkState.StatusCode(), ShouldEqual, 200)
+			})
+
+		})
+	})
+}
+
+func TestCheckStatus204(t *testing.T) {
+	Convey("Given that the vault client is available", t, func() {
+		mockedNomad := getMockNomad(
+			http.Request{Method: "GET"},
+			MockedHTTPResponse{StatusCode: 204, Body: `{"status": "OK"}`})
+
+		Convey("When a check is performed", func() {
+			checkState := health.NewCheckState(nomad.ServiceName)
+
+			Convey("Then the Checker returns no error", func() {
+				err := mockedNomad.Checker(ctx, checkState)
+				So(err, ShouldBeNil)
+				So(checkState.StatusCode(), ShouldEqual, 204)
```

Comment:
given that the other tests are checking the .Status() I would expect this to as well. Especially because I think this would be critical? Only a 200 is a 'good' response from nomad, so the checker should be setting a critical status when a 204 comes back

Reply:
Or if the nomad endpoint genuinely might return a 204 meaning all is fine, the if statement in the actual code should be changed, and this code made more specific to highlight that